### PR TITLE
Added a layer of checking on Node

### PR DESCRIPTION
**Pull request by mmoreram**
* Added some tests covering these methods

### CODE REVIEW

**Comment by mmoreram** [comment]
@EmanueleMinotto related to #1008

File: src/Elcodi/Component/Menu/Entity/Menu/Node.php [hunk 1]
```diff
@@ -290,6 +297,10 @@ public function isActive($currentUrl)
      */
     public function isExpanded($currentUrl)
     {
+        if (!($this->subnodes instanceof Collection)) {
```
Comment (EmanueleMinotto) [inline src/Elcodi/Component/Menu/Entity/Menu/Node.php]: You could also remove the internal parentheses: `if (!$this->subnodes instanceof Collection) {`

Comment (mmoreram) [inline src/Elcodi/Component/Menu/Entity/Menu/Node.php]: Are you sure of that?
first is `!$this->subnodes`
then `{result} instanceof Collection

Comment (EmanueleMinotto) [inline src/Elcodi/Component/Menu/Entity/Menu/Node.php]: check the VLD: https://3v4l.org/MJuCR/vld#tabs
the `instanceof` has a higher precedence than `!` and both of them give the same output

Comment (mmoreram) [inline src/Elcodi/Component/Menu/Entity/Menu/Node.php]: Good then :D Thanks for that info! ^^ I was very used of working with these parenthesis :dancer:

**Pull request merged**